beep/protocol_tools/arbin_schedule_file.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2019 Toyota Research Institute. All rights reserved.
+""" Schedule file parsing and parameter insertion"""
+
+
+import os
+import hashlib
+import re
+from beep import SCHEDULE_TEMPLATE_DIR
+from collections import OrderedDict, defaultdict
+
+
+class ScheduleFile:
+    """
+    Schedule file utility. Provides the ability to read an Arbin type schedule file
+
+    Args:
+        version (str): Default version.
+        section_regex (raw str): regex string to return all section headers from
+            the schedule file
+        step_regex (raw str): regex string to return all step headers from the
+            schedule file
+        limit_regex (raw str): regex string to return all limit headers from
+            the schedule file
+
+    """
+    def __init__(self,
+                 version=None,
+                 section_regex=r'(?<=\[).*',
+                 step_regex=r'.Schedule_Step[0-9]*',
+                 limit_regex=r'.Schedule_Step[0-9]*_Limit[0-9]*'):
+        self.service = version
```

Comment:
Just wondering where the term "service", instead of version, comes from, both here and in ProcedureToSchedule.

Reply:
Yeah, we were trying to include some kind of versioning of the service, but I think this is now obsolete since the entire package is version controlled. I will clean this up.